python/tk_nuke/context.py

```diff
@@ -0,0 +1,373 @@
+# Copyright (c) 2013 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import sys
+import copy
+
+import nuke
+import tank
+
+from tank import TankError
+
+class StudioContextSwitcher(object):
+    """
+    A Toolkit context-switching manager.
+    """
+
+    def __init__(self, engine):
+        """
+        Initializes a StudioContextSwitcher object.
+
+        :param engine:  The running sgtk.engine.Engine to associate the
+                        context switcher with.
+        """
+        self._event_desc = [
+            dict(
+                add=nuke.addOnCreate,
+                remove=nuke.removeOnCreate,
+                registrar=nuke.callbacks.onCreates,
+                function=self._startup_node_callback,
+            ),
+            dict(
+                add=nuke.addOnScriptSave,
+                remove=nuke.removeOnScriptSave,
+                registrar=nuke.callbacks.onScriptSaves,
+                function=self._on_save_callback,
+            ),
+        ]
+
+        self._engine = engine
+        self._context_cache = dict()
+        self._init_project_root = engine.tank.project_path
+        self._init_context = engine.context
+        self._current_studio_context = False
+
+        self.register_events(reregister=True)
+
+    ##########################################################################
+    # properties
+
+    @property
+    def context(self):
+        """
+        The current sgtk.context.Context.
+        """
+        if self.engine:
+            self._context = self.engine.context
+        return self._context
+
+    @property
+    def current_studio_context(self):
+        """
+        Whether Nuke Studio is in "Hiero" or "Nuke" mode presently.
+        """
+        return ['Hiero', 'Nuke'][self._current_studio_context]
```

Comment:
To be clear, what I mean is that I feel that _current_studio_context is about to tell me what the current studio context is. It sounds like I should be expecting a name, but instead I am getting a boolean.